luigi/task.py

```diff
@@ -484,6 +484,15 @@ def complete(self):
 
         return all(itertools.imap(lambda output: output.exists(), outputs))
 
+    @classmethod
+    def bulk_complete(cls, parameter_tuples):
+        """Returns those of parameter_tuples for which this Task is complete.
+
+        Override (with an efficient implementation) for efficient scheduling
+        with range tools. Keep the logic consistent with that of complete().
+        """
+        raise NotImplementedError
```

Comment:
1. I'm generally a bit skeptical about adding stuff to the base Task class since it makes the interface more complicated. But I guess in this case it makes sense since it's such a big performance optimization

2. The default behavior could be to return `all([cls(*parameter).complete for parameter in parameter_tuples])` instead of raising a `NotImplementedError`. I would have a mild preference towards doing that.